pkg/cluster/k8sres.go

```diff
@@ -329,6 +329,20 @@ func tolerations(tolerationsSpec *[]v1.Toleration, podToleration map[string]stri
 	return []v1.Toleration{}
 }
 
+func nodeSelector(clusterManifestSelector map[string]string, operatorSelectorConfig map[string]string) map[string]string {
```

Comment:
we already have the func `nodeAffinity` - you could extend it to also take the `nodeSelector` entries from the manifest. Then generateStatefulset and generatePodTemplate don't need more arguments. That means for the e2e test you have to look `nodeAffinity` and not `nodeSelector`.